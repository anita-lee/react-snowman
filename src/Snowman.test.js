import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Snowman from "./Snowman";

it("renders without crashing", function() {
  render(<Snowman />);
});

it("matches snapshot", function() {
  const { container } = render(<Snowman />);
  expect(container).toMatchSnapshot();
});

it("after 6 wrong guesses: no buttons shown, 'You lose' and show correct word", function() {
  const { container, debug } = render(<Snowman />);

  const buttonB = container.querySelector(`.Snowman-letter-b`);
  const buttonC = container.querySelector(`.Snowman-letter-c`);
  const buttonD = container.querySelector(`.Snowman-letter-d`);
  const buttonF = container.querySelector(`.Snowman-letter-f`);
  const buttonG = container.querySelector(`.Snowman-letter-g`);
  const buttonH = container.querySelector(`.Snowman-letter-h`);

  fireEvent.click(buttonB);
  fireEvent.click(buttonC);
  fireEvent.click(buttonD);
  fireEvent.click(buttonF);
  fireEvent.click(buttonG);
  fireEvent.click(buttonH);
  debug();

  //FIXME: className of button area is not null, it now has You lose and apple in it.
  expect(container.querySelector(".Snowman-letters")).toBeNull();

  expect(container.toBeInTheDocument("You lose"));
  expect(container.toBeInTheDocument("apple"));
});

// it("after 6 wrong guesses: no buttons shown", function() {
//   const { container } = render(<Snowman />);



//   expect(container.querySelector(".Snowman-letter-button")).toBeNull();
// });

// it("counts correctly when heads appears", function() {
//   const { container } = render(<CoinContainer />);

//   const button = container.querySelector("button");
//   fireEvent.click(button);

//   expect(container.querySelector("img[alt='head']")).toBeInTheDocument();
//   expect(container.querySelector("img[alt='tail']")).not.toBeInTheDocument();
//   expect(container.querySelector("p"))
//     .toHaveTextContent("Out of 1 flips, there have been 1 heads and 0 tails.");
// });

// it("counts correctly when tails appears", function() {
//   const { container } = render(<CoinContainer />);

//   const button = container.querySelector("button");
//   fireEvent.click(button);
//   fireEvent.click(button);

//   expect(container.querySelector("img[alt='head']")).not.toBeInTheDocument();
//   expect(container.querySelector("img[alt='tail']")).toBeInTheDocument();
//   expect(container.querySelector("p"))
//     .toHaveTextContent("Out of 2 flips, there have been 1 heads and 1 tails.");
// });

// afterEach(function() {
//   Math.random.mockRestore();
// });
