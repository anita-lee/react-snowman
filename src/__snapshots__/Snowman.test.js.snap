// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Snowman"
  >
    <p>
      Number wrong: 
      0
    </p>
    <img
      alt="0"
      src="0.png"
    />
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
      _
    </p>
    <p
      class="Snowman-letters"
    >
      <button
        class="Snowman-letter-a"
        value="a"
      >
        a
      </button>
      <button
        class="Snowman-letter-b"
        value="b"
      >
        b
      </button>
      <button
        class="Snowman-letter-c"
        value="c"
      >
        c
      </button>
      <button
        class="Snowman-letter-d"
        value="d"
      >
        d
      </button>
      <button
        class="Snowman-letter-e"
        value="e"
      >
        e
      </button>
      <button
        class="Snowman-letter-f"
        value="f"
      >
        f
      </button>
      <button
        class="Snowman-letter-g"
        value="g"
      >
        g
      </button>
      <button
        class="Snowman-letter-h"
        value="h"
      >
        h
      </button>
      <button
        class="Snowman-letter-i"
        value="i"
      >
        i
      </button>
      <button
        class="Snowman-letter-j"
        value="j"
      >
        j
      </button>
      <button
        class="Snowman-letter-k"
        value="k"
      >
        k
      </button>
      <button
        class="Snowman-letter-l"
        value="l"
      >
        l
      </button>
      <button
        class="Snowman-letter-m"
        value="m"
      >
        m
      </button>
      <button
        class="Snowman-letter-n"
        value="n"
      >
        n
      </button>
      <button
        class="Snowman-letter-o"
        value="o"
      >
        o
      </button>
      <button
        class="Snowman-letter-p"
        value="p"
      >
        p
      </button>
      <button
        class="Snowman-letter-q"
        value="q"
      >
        q
      </button>
      <button
        class="Snowman-letter-r"
        value="r"
      >
        r
      </button>
      <button
        class="Snowman-letter-s"
        value="s"
      >
        s
      </button>
      <button
        class="Snowman-letter-t"
        value="t"
      >
        t
      </button>
      <button
        class="Snowman-letter-u"
        value="u"
      >
        u
      </button>
      <button
        class="Snowman-letter-v"
        value="v"
      >
        v
      </button>
      <button
        class="Snowman-letter-w"
        value="w"
      >
        w
      </button>
      <button
        class="Snowman-letter-x"
        value="x"
      >
        x
      </button>
      <button
        class="Snowman-letter-y"
        value="y"
      >
        y
      </button>
      <button
        class="Snowman-letter-z"
        value="z"
      >
        z
      </button>
       
    </p>
  </div>
</div>
`;
